pydantic/json_schema.py

```diff
@@ -1064,24 +1065,32 @@ def field_is_present(self, field: CoreSchemaField) -> bool:
             assert_never(self.mode)
 
     def field_is_required(
-        self, field: core_schema.ModelField | core_schema.DataclassField | core_schema.TypedDictField
+        self,
+        field: core_schema.ModelField | core_schema.DataclassField | core_schema.TypedDictField,
+        total: bool,
     ) -> bool:
         """Whether the field should be marked as required in the generated JSON schema.
         (Note that this is irrelevant if the field is not present in the JSON schema.).
+
+        Args:
+            field: The schema for the field itself.
+            total: Only applies to `TypedDictField`s.
+                Indicates if the `TypedDict` this field belongs to is total, in which case any fields that don't
+                explicitly specify `required=False` are required.
```

Comment:
You may can add `Returns` and `Raises` section to docstring.

Reply:
Seems overkill to me